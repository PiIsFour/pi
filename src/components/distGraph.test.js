import React from 'react'
import { shallow } from 'enzyme'
import enzymeToJson from 'enzyme-to-json'

import DistGraph from './distGraph'

describe('distGraph', () => {
	it('snapshot matches', () => {
		const dist = [
			{param: 0, prob: 0.25},
			{param: 0.3, prob: 0.5},
			{param: 1, prob: 0.25}
		]
		const wrapper = shallow(<DistGraph dist={dist}/>)
		expect(enzymeToJson(wrapper)).toMatchSnapshot()
	})

	it('renders a svg', () => {
		const dist = [
			{param: 0, prob: 0.25},
			{param: 0.3, prob: 0.5},
			{param: 1, prob: 0.25}
		]
		const wrapper = shallow(<DistGraph dist={dist}/>)
		expect(wrapper).toHaveTagName('svg')
	})

	it('has a path with the right instructions', () => {
		const dist = [
			{param: 0, prob: 0.25},
			{param: 0.3, prob: 0.5},
			{param: 1, prob: 0.25}
		]
		const wrapper = shallow(<DistGraph dist={dist}/>)
		expect(wrapper.find('path')).toHaveProp('d', 'M 0 50 L 60 0 L 200 50')
	})
})
