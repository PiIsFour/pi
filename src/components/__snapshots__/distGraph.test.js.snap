// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`distGraph snapshot matches 1`] = `
<svg
  viewBox="-10 -5 220 110"
>
  <path
    d="M 0 50 L 60 0 L 200 50"
    fill="transparent"
    stroke="black"
    stroke-width="0.5"
  />
</svg>
`;
